azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/resources/templates.json

```diff
@@ -451,6 +451,113 @@
+            "direction": "out",
+            "name": "pubEvent",
+            "pubsubname": "messagebus",
+            "topic": "B"
+          },
+          {
+            "type": "daprTopicTrigger",
+            "pubsubname": "messagebus",
+            "topic": "A",
+            "name": "subEvent",
+            "direction": "in",
+            "dataType": "string"
+          }
+        ]
+      },
+      "files": {
+        "function.java": "package com.function;\r\n\r\nimport com.fasterxml.jackson.core.JsonProcessingException;\r\nimport com.fasterxml.jackson.databind.JsonNode;\r\nimport com.fasterxml.jackson.databind.ObjectMapper;\r\nimport com.microsoft.azure.functions.ExecutionContext;\r\nimport com.microsoft.azure.functions.OutputBinding;\r\nimport com.microsoft.azure.functions.annotation.FunctionName;\r\nimport com.microsoft.azure.functions.dapr.annotation.DaprStateOutput;\r\nimport com.microsoft.azure.functions.dapr.annotation.DaprTopicTrigger;\r\n\r\n\r\npublic class DaprTopicTriggerFunc {\r\n\r\n    \/*\r\n    Visit https:\/\/aka.ms\/azure-functions-dapr to learn how to use the Dapr extension.\r\n    These tasks should be completed prior to running :\r\n      1. Install Dapr\r\n      2. Change the bundle name in host.json to \"Microsoft.Azure.Functions.ExtensionBundle.Preview\" and the version to \"[4.*, 5.0.0)\"\r\n    Start function app with Dapr: dapr run --app-id functionapp --app-port 3001 --dapr-http-port 3501 --components-path .\/components\/ -- mvn azure-functions:run\r\n    Invoke function app: dapr publish --pubsub messagebus --publish-app-id functionapp --topic A --data '{\\\"value\\\": { \\\"orderId\\\": \\\"42\\\" } }'\r\n    *\/\r\n    @FunctionName(\"DaprTopicTriggerFunc\")\r\n    public String run(\r\n        @DaprTopicTrigger(\r\n            pubSubName = \"messagebus\",\r\n            topic = \"X\")\r\n            String request,\r\n        @DaprStateOutput(\r\n            stateStore = \"statestore\",\r\n            key = \"product\")\r\n        OutputBinding<String> product,\r\n        final ExecutionContext context) throws JsonProcessingException {\r\n        context.getLogger().info(\"Java DaprTopic trigger with DaprState output binding function processed a request from the Dapr Runtime.\");\r\n\r\n        \/\/ Get the CloudEvent data from the request body as a JSON string\r\n        ObjectMapper objectMapper = new ObjectMapper();\r\n        JsonNode jsonNode = objectMapper.readTree(request).get(\"data\");\r\n        String serializedJson = objectMapper.writeValueAsString(jsonNode);\r\n\r\n        product.setValue(serializedJson);\r\n\r\n        return serializedJson;\r\n    }\r\n}",
+        "components\\cron_bindings.yaml": "apiVersion: dapr.io\/v1alpha1\r\nkind: Component\r\nmetadata:\r\n  name: scheduled\r\nspec:\r\n  type: bindings.cron\r\n  version: v1\r\n  metadata:\r\n  - name: schedule\r\n    value: \"@every 10s\" # valid cron schedule\r\n  - name: direction\r\n    value: \"input\"",
```

Comment:
currently we do not support create multiple files with function java template

Reply:
We have added this item to our back log, as a workaround, you may also add the configuration files to maven archetype, and add them with groovy script here https://github.com/microsoft/azure-maven-archetypes/blob/develop/azure-functions-archetype/src/main/resources/META-INF/archetype-post-generate.groovy

Reply:
I think it will become cumbersome if we add Dapr specific files with multiple if conditions. Because all three templates have different component files. Do we have any ETA for the backlog item?

Reply:
@ASHIQUEMD Thanks for your efforts and sorry for the late response, we have prioritized this working item, which should be done in our October release.